### PR TITLE
Fix scheduler race around restaring tasks

### PR DESCRIPTION
**Pull request by dmzmk**
Scheduler now tracks both pipe and restart timer handles and properly cancels either of them

### CODE REVIEW

File: pipe/runtime/src/scheduler.rs [hunk 1]
```diff
@@ -177,6 +213,7 @@ where
             None => ScheduleResult::New,
         };
         self.pipe_configs.insert(id, config);
+        self.unschedule(id).await;
```
Comment (wwkeyboard) [inline pipe/runtime/src/scheduler.rs:216]: Why does this unschedule and then reschedule here?

**Review by wwkeyboard — APPROVED** [review]
Aside from the question about the extra unschedule this looks good to me

File: pipe/runtime/src/scheduler.rs (hunk 1 above)
Comment (dmzmk) [reply pipe/runtime/src/scheduler.rs:216]: unschedule makes sure that current pipe handle (if any) is definitely aborted, i.e. if pipe in restart state - timer should never fire if we re-added pipe

Comment (dmzmk) [reply pipe/runtime/src/scheduler.rs:216]: @wwkeyboard let me know if this call is still a concern

**Pull request merged**